ui/components/ui/nft-info-setApprovalForAll/nft-info-setApprovalForAll.js

```diff
@@ -0,0 +1,157 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import { I18nContext } from '../../../contexts/i18n';
+import Box from '../box';
+import Typography from '../typography';
+import {
+  COLORS,
+  DISPLAY,
+  FONT_WEIGHT,
+  TYPOGRAPHY,
+  TEXT_ALIGN,
+} from '../../../helpers/constants/design-system';
+import Identicon from '../identicon';
+import Tooltip from '../tooltip/tooltip';
+import { getTokenList } from '../../../selectors';
+
+export default function NftInfoSetApprovalForAll({
+  assetName,
+  tokenAddress,
+  total,
+  collections = {},
+}) {
+  const t = useContext(I18nContext);
+  const tokenList = useSelector(getTokenList);
+
+  const nftTokenListImage = tokenList[tokenAddress.toLowerCase()]?.iconUrl;
+
+  let nftCollectionNameExist;
+  let nftCollectionImageExist;
+
+  Object.values(collections).forEach((nftCollections) => {
+    if (nftCollections.collectionName === assetName) {
+      nftCollectionNameExist = nftCollections.collectionName;
+      nftCollectionImageExist = nftCollections.collectionImage;
+    }
+  });
+
+  const renderCollectionImage = (collectionImage, collectionName, key) => {
+    if (collectionImage) {
+      return <Identicon diameter={24} image={collectionImage} />;
+    }
+    return (
+      <Box
+        key={key}
+        color={COLORS.OVERLAY_INVERSE}
+        textAlign={TEXT_ALIGN.CENTER}
+        className="nft-info-setApproveForAll__collection-image-alt"
+      >
+        {collectionName?.[0]?.toUpperCase() ?? null}
+      </Box>
+    );
+  };
+
+  return (
+    <Box
+      display={DISPLAY.FLEX}
+      className="nft-info-setApproveForAll"
+      backgroundColor={COLORS.BACKGROUND_ALTERNATIVE}
+    >
+      <Box marginTop={4} marginBottom={4} marginLeft={4}>
+        <Typography
+          fontWeight={FONT_WEIGHT.BOLD}
+          variant={TYPOGRAPHY.H6}
+          marginTop={0}
+        >
+          {t('nftCollectionName')}:
+        </Typography>
+        <Box display={DISPLAY.FLEX}>
+          <Box marginBottom={4}>
+            {Object.keys(collections).length > 0 && nftCollectionNameExist
```

Comment:
There is something here we can improve on a little bit. Above we iterate over `collections` values to determine if `nftCollectionNameExist` and then here we look at the keys of `collections` and use the length of that + `nftCollectionNameExist` to decide which parameters too pass to `renderCollectionImage`

Also, above we are iterating over the values of an object to check if the `collectionName` property of one of those objects matches the `assetName` passed to this component.

I wonder if we can simplify things a bit. What is `collections` keyed by? Is it keyed by the asset name? Or perhaps by an id that can be passed down from a parent? If so, we could just do something like `nftCollectionNameExist = collections[id].collectionName; nftCollectionImageExist = collections[id].collectionImage;`

Then, if we can do that, we can go a step further and perhaps refactor this section of the code to
```
<Box marginBottom={4}>
    {renderCollectionImageOrName(collectioon, assetName, id)}
</Box>
```
And then have all the logic contained within that function (instead of some in the function, nftTokenListImage defined outside the function and then some logic here)

Reply:
Done [here](https://github.com/MetaMask/metamask-extension/pull/15844/files#diff-ec98920531d74df41855f0a46de3c187b84296d0ca2a6cc2e67548d48db837e9).